cpp/csp/engine/RootEngine.cpp

```diff
@@ -11,12 +11,12 @@
 namespace csp
 {
 
-static volatile bool    g_SIGNALED = false;
+static volatile int     g_SIGNAL_COUNT = 0;
```

Comment:
Can we add a comment explaining why this was made a counter

Reply:
Done